### ISSUE TITLE
BUILD FAILURE running -PautoInstallPackage -Padobe-public clean install

### ISSUE THREAD

**Issue by jessicazink** [adobe/aem-guides-wknd#195]
Expected Behaviour: Build Successful 

Actual Behaviour:  WKND Sites Project - UI Tests ...................... FAILURE 
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single (make-assembly) on project aem-guides-wknd.ui.tests: Execution make-assembly of goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single failed: group id '2099151705' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit

Steps to Reproduce: Run -PautoInstallPackage -Padobe-public clean install in Visual Studio
I get this error each time I try to run a clean install

Platform and Version
AEM 6.5
Java version 11.0.10
Apache Maven 3.6.3 

Log/Results
Reactor Summary for aem-guides-wknd 0.0.1-SNAPSHOT:
[INFO] 
[INFO] aem-guides-wknd .................................... SUCCESS [  0.201 s]
[INFO] WKND Sites Project - Core .......................... SUCCESS [  6.056 s]
[INFO] WKND Sites Project - UI Frontend ................... SUCCESS [ 11.092 s]
[INFO] WKND Sites Project - Repository Structure Package .. SUCCESS [  0.956 s]
[INFO] WKND Sites Project - UI apps ....................... SUCCESS [  4.886 s]
[INFO] WKND Sites Project - UI content .................... SUCCESS [  2.653 s]
[INFO] WKND Sites Project - UI config ..................... SUCCESS [  0.396 s]
[INFO] WKND Sites Project - All ........................... SUCCESS [  0.318 s]
[INFO] WKND Sites Project - Integration Tests ............. SUCCESS [  5.888 s]
[INFO] WKND Sites Project - Dispatcher .................... SUCCESS [  0.063 s]
[INFO] WKND Sites Project - UI Tests ...................... FAILURE [  0.191 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  33.610 s
[INFO] Finished at: 2021-02-15T15:14:08-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single (make-assembly) on project aem-guides-wknd.ui.tests: Execution make-assembly of goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single failed: group id '2099151705' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :aem-guides-wknd.ui.tests

**Comment by davidrmckinnon** [comment]
Hi @jessicazink ,

I have the same issue and found a fix in this Stackoverflow answer
https://stackoverflow.com/questions/33819438/maven-assembly-plugin-group-id-1377585961-is-too-big-error

The fix is to change `<tarLongFileMode>gnu</tarLongFileMode>` in ui.tests/pom.xml line 100
to `<tarLongFileMode>posix</tarLongFileMode>` 

That is, change
```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-assembly-plugin</artifactId>
    <configuration>
        <descriptors>
            <descriptor>${project.basedir}/assembly-ui-test-docker-context.xml</descriptor>
        </descriptors>
        <tarLongFileMode>gnu</tarLongFileMode>
    </configuration>
    <executions>
```

to
```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-assembly-plugin</artifactId>
    <configuration>
        <descriptors>
            <descriptor>${project.basedir}/assembly-ui-test-docker-context.xml</descriptor>
        </descriptors>
        <tarLongFileMode>posix</tarLongFileMode>
    </configuration>
    <executions>
```

Hope that helps.

**Comment by godanny86** [comment]
hey @davidrmckinnon , @jessicazink what OS are you using?

**Comment by jessicazink** [comment]
@davidrmckinnon that fixed it! Thank you. @godanny86 I'm using Mac OSCatalina version 10.15.7 for reference

**Comment by davidrmckinnon** [comment]
No worries @jessicazink

@godanny86 I'm using the same as @jessicazink: macOS 10.15.7

I also noticed this in the Maven Assembly Plugins FAQ

> Tar complains about groupid value being too big
GNU tar has restrictions on max value of UID/GUID in tar files. Consider using the more well defined POSIX tar format, which should support larger values. Use tarLongFileMode=posix in the assembly:single goal.

https://maven.apache.org/plugins/maven-assembly-plugin/faq.html#tarFileModes

Might be worth making the switch permanently?

**Comment by godanny86** [comment]
Thanks @davidrmckinnon. Yup agreed we might need to make this switch permanently. 

Just curious:
1. Do you get the same issue if you drop the `-Pclassic` profile?
2. If you generate a project using the AEM Project archetype https://github.com/adobe/aem-project-archetype and then build the project, same issue?

**Comment by davidrmckinnon** [comment]
1. Yes, with `<tarLongFileMode>gnu</tarLongFileMode>` the build fails at the same point using `mvn clean install -PautoInstallSinglePackage` and `mvn clean install -PautoInstallSinglePackage -Pclassic`
2. No, the archetype builds just fine with `<tarLongFileMode>gnu</tarLongFileMode>`, which seems odd.

This is the properties I used for the Archetype:
```
groupId=com.tst
artifactId=tst
version=1.0.0-SNAPSHOT
package=com.tst
appId=tst
appTitle=Archetype 25 Test
aemVersion=6.5.7
sdkVersion=latest
languageCountry=${languageCountry}
includeExamples=y
includeErrorHandler=n
frontendModule=general
singleCountry=y
includeDispatcherConfig=y
includeCommerce=n
includeForms=n
sdkFormsVersion=latest
commerceEndpoint=https://hostname.com/graphql
datalayer=y
amp=n
enableDynamicMedia=n
```

**Comment by godanny86** [comment]
Thanks @davidrmckinnon for following up. I'll try and compare the two. This project is based on the archetype but parts of it have changed to make it work for both Cloud Service and 6.5/6.4

**Comment by godanny86** [comment]
hey @davidrmckinnon and @jessicazink if you have a few minutes, mind testing this branch: https://github.com/adobe/aem-guides-wknd/tree/ISSUE-195/ui-test-failure ?

**Comment by davidrmckinnon** [comment]
Hi @godanny86 ,

I ran both `mvn clean install -PautoInstallSinglePackage -Pclassic` and `mvn clean install -PautoInstallSinglePackage`

Both builds were successful. No errors.

**Comment by jessicazink** [comment]
@godanny86 I'll try to run this today, but I'm just reaching the end of the WKND tutorial and I'm worried about losing the work I've done so far. I'm really new to maven.

**Comment by godanny86** [comment]
@jessicazink, no problem! I think having @davidrmckinnon's testing should be good. I've had a few folks internally test as well. Thank you both!

**Comment by jessicazink** [comment]
@godanny86 Thank you!

**Comment by godanny86** [comment]
closing this, as it should be fixed with 0.2.0 release. Feel free to re-open if you see other issues.
